src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -488,26 +482,62 @@ pal::string_t fx_muxer_t::resolve_fx_dir(host_mode_t mode,
             }
         }
 
+        fx_ver_t resolved_ver = specified;
         if (!do_roll_forward)
         {
             trace::verbose(_X("Did not roll forward because specified version='%s', patch_roll_fwd=%d, roll_fwd_on_no_candidate_fx=%d, chose [%s]"),
                 specified_fx_version.c_str(), patch_roll_fwd_enabled, roll_fwd_on_no_candidate_fx_val, fx_ver.c_str());
-            append_path(&fx_dir, fx_ver.c_str());
         }
         else
         {
-            resolve_roll_forward(fx_dir, fx_ver, specified, patch_roll_fwd_enabled, roll_fwd_on_no_candidate_fx_val);
+            std::vector<pal::string_t> list;
+            std::vector<fx_ver_t> version_list;
+            pal::readdir(fx_dir, &list);
+
+            for (const auto& version : list)
+            {
+                fx_ver_t ver(-1, -1, -1);
+                if (fx_ver_t::parse(version, &ver, false))
+                {
+                    version_list.push_back(ver);
+                }
+            }
+            resolved_ver = resolve_most_compatible(version_list, fx_ver, specified, patch_roll_fwd_enabled, roll_fwd_on_no_candidate_fx_val);
         }
 
+        pal::string_t resolved_ver_str = resolved_ver.as_str();
+        append_path(&fx_dir, resolved_ver_str.c_str());
+
         if (pal::directory_exists(fx_dir))
         {
             trace::verbose(_X("Chose FX version [%s]"), fx_dir.c_str());
-            return fx_dir;
+
+            if (!do_roll_forward)
```

Comment:
This check looks incorrect since for a given dir in the hiv_dir you enumerated above, resolve_most_compatible has already performed patch roll forward if it was applicable.